In [1]:
import pandas as pd
import statsmodels.api as sm
from patsy import dmatrices
import datetime

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [2]:
df = pd.read_csv('/content/day.csv')


### 1. Obtenha um modelo a partir da função statsmodels.api

In [3]:
df.tail()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
726      727  2012-12-27       1   1    12        0        4           1   
727      728  2012-12-28       1   1    12        0        5           1   
728      729  2012-12-29       1   1    12        0        6           0   
729      730  2012-12-30       1   1    12        0        0           0   
730      731  2012-12-31       1   1    12        0        1           1   

     weathersit      temp     atemp       hum  windspeed  casual  registered  \
726           2  0.254167  0.226642  0.652917   0.350133     247        1867   
727           2  0.253333  0.255046  0.590000   0.155471     644        2451   
728           2  0.253333  0.242400  0.752917   0.124383     159        1182   
729           1  0.255833  0.231700  0.483333   0.350754     364        1432   
730           2  0.215833  0.223487  0.577500   0.154846     439        2290   

      cnt  
726  2114  
727  3095  
728  1341  
729  1796  
730  2729

In [4]:
df['dteday'] = pd.to_datetime(df['dteday'])

In [5]:
vars = ['season','weekday','holiday','workingday','weathersit','temp','atemp','hum','windspeed','cnt'] # Variaveis que julgo mais impactantes no aluguel de bicicletas
df_variaveis = df[vars]

In [6]:
y, X = dmatrices('cnt ~ season + weekday + holiday + workingday + weathersit + temp + atemp + hum + windspeed', data=df_variaveis, return_type='dataframe')

In [7]:
modelo = sm.OLS(y,X)
resultado = modelo.fit()

In [8]:
print(resultado.summary())

                            OLS Regression Results                            
Dep. Variable:                    cnt   R-squared:                       0.527
Model:                            OLS   Adj. R-squared:                  0.521
Method:                 Least Squares   F-statistic:                     89.31
Date:                Mon, 18 Apr 2022   Prob (F-statistic):          4.63e-111
Time:                        20:24:30   Log-Likelihood:                -6295.9
No. Observations:                 731   AIC:                         1.261e+04
Df Residuals:                     721   BIC:                         1.266e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3037.1604    360.990      8.413      0.0

### 2. Escreva a equação da regressão estimada

cnt = 3037.1604 + (*season* * 405.6633) + (weekday * 60.3889) + (*holiday* * -495.7224) + (workingday * 113.9386) + (*weathersit* * -490.2313) + (temp * 2370.0184) + (*atemp* *  3692.6385) + (hum * -2286.0650) + (*windspeed* * -3131.5379)

### 3. Escreva o valor de r2

R-squared:                       0.527
